source/compiler/qsc_qasm/src/tests/statement/pragma/box.rs

```diff
@@ -0,0 +1,221 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use crate::tests::compile_qasm_to_qsharp;
+use expect_test::expect;
+use miette::Report;
+
+#[test]
+fn pragma_target_can_be_defined_before_pragma() -> miette::Result<(), Vec<Report>> {
+    let source = r#"
+        def sample_box_target() {}
+        pragma qdk.box.open sample_box_target
+        box {}
+    "#;
+
+    let qsharp = compile_qasm_to_qsharp(source)?;
+    expect![[r#"
+        import Std.OpenQASM.Intrinsic.*;
+        function sample_box_target() : Unit {}
+        {
+            sample_box_target();
+        };
+    "#]]
+    .assert_eq(&qsharp);
+    Ok(())
+}
+
+#[test]
+fn pragma_target_can_be_defined_after_pragma() -> miette::Result<(), Vec<Report>> {
+    let source = r#"
+        pragma qdk.box.open sample_box_target
+        def sample_box_target() {}
+        box {}
+    "#;
+
+    let qsharp = compile_qasm_to_qsharp(source)?;
+    expect![[r#"
+        import Std.OpenQASM.Intrinsic.*;
+        function sample_box_target() : Unit {}
+        {
+            sample_box_target();
+        };
+    "#]]
+    .assert_eq(&qsharp);
+    Ok(())
+}
+
+#[test]
+fn pragma_target_can_be_used_by_multilple_pragmas() -> miette::Result<(), Vec<Report>> {
+    let source = r#"
+        pragma qdk.box.open sample_box_target
+        pragma qdk.box.close sample_box_target
+        def sample_box_target() {}
+        box {}
+    "#;
+
+    let qsharp = compile_qasm_to_qsharp(source)?;
+    expect![[r#"
+        import Std.OpenQASM.Intrinsic.*;
+        function sample_box_target() : Unit {}
+        {
+            sample_box_target();
+            sample_box_target();
+        };
+    "#]]
+    .assert_eq(&qsharp);
+    Ok(())
+}
+
+#[test]
+fn pragmas_can_have_separate_targets() -> miette::Result<(), Vec<Report>> {
```

Comment:
Might be worth adding a test that sets two pragmas for box open/close and confirms only the last one "wins"